src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,8 +348,23 @@ export class DataAccessorBasedStore implements ResourceStore {
    * @param slug - Slug to use for the new URI.
    */
   protected createURI(container: ResourceIdentifier, isContainer: boolean, slug?: string): ResourceIdentifier {
-    return { path:
-        `${ensureTrailingSlash(container.path)}${slug ? trimTrailingSlashes(slug) : uuid()}${isContainer ? '/' : ''}` };
+    const base = ensureTrailingSlash(container.path);
+    const cleanedSlug = slug ? this.cleanSlug(slug) : null;
+    const name = cleanedSlug ?? uuid();
+    const suffix = isContainer ? '/' : '';
+    return { path: `${base}${name}${suffix}` };
+  }
+
+  /**
+   * Clean http Slug to be compatible with the server. Makes sure there are no unwanted characters
+   * e.g.: cleanslug('&/%26') returns '%26%2F%26'
+   * @param slug - the slug to clean
+   */
+  protected cleanSlug(slug: string): string {
+    if (/\/[^/]/u.test(slug)) {
```

Comment:
As mentioned, I would just disallow all slash occurrences. So this can be replaced with `slug.includes('/')`.

Reply:
using `slug.includes('/')` will error when trying to create a folder via POST. Is this the desired outcome?

Reply:
The creation of a folder by a client should be indicated via other headers, right?

Reply:
Ah yea that's true. You can leave that for now then, we can always change later if we decide not to want that.

Reply:
> The creation of a folder by a client should be indicated via other headers, right?

I'm ok with either since I don't know what the current accepted standard is in NSS for example. It is possible to create containers without header by doing a PUT to `/foo/` for example so it would not be that inconsistent to allow this.

Reply:
> I'm ok with either since I don't know what the current accepted standard is

https://www.w3.org/TR/ldp/#ldpc-post-createrdf

So as far as I understand, client can (only) create containers by having a `Link` header for a basic container. We might want to resolve that in a new issue. @csarven?